### ISSUE TITLE
`url.format()` formats empty array to be empty query parameter (`a=&&c=`)

### ISSUE THREAD

**Issue by joshuali925** [defunctzombie/node-url#72]
Reproduce steps:
1. go to https://playcode.io/react
2. on the left, add package `url` with latest version `0.11.4`
3. modify code on the right
```tsx
import React from "react";
import { format } from "url";

export function App(props) {
  return (
    <div className="App">
      <h1>
        {format({
          pathname: "/",
          query: {
            a: 1,
            b: [],
            c: 1,
          },
        })}
      </h1>
    </div>
  );
}
```
4. see rendered result: `/?a=1&&c=1`

But in nodejs, this will be 
```sh
❯ node -e 'console.log(require("url").format({ pathname: "/", query: { a: 1, b: [], c: 1, }, }))'
/?a=1&c=1
```

node version is v18.19.0

Is this expected behavior?

**Comment by ljharb** [comment]
This does seem like a bug; even down to node 0.10, your snippet does give that output.

**Comment by joshuali925** [comment]
node 0.10.0 indeed gives `/?a=1&&c=1`, but i tried node >= 0.11.14, the output is `/?a=1&c=1`
```sh
❯ asdf shell nodejs 0.11.14
❯ node -e 'console.log(require("url").format({ pathname: "/", query: { a: 1, b: [], c: 1, }, }))'
/?a=1&c=1

❯ asdf shell nodejs 0.11.13
❯ node -e 'console.log(require("url").format({ pathname: "/", query: { a: 1, b: [], c: 1, }, }))'
/?a=1&&c=1
```

it seems to be a bug that got fixed on nodejs side

**Comment by ljharb** [comment]
aha, i missed the double ampersands. This package currently matches a very old version of node, but since we're at version 0.11, matching 0.11 latest is an obvious goal. Let's fix the bug here too :-)

**Comment by joshuali925** [comment]
I got some more time to look into this, it seems https://playcode.io/react is misleading.. We are using url version 0.10.3 which has the issue, and using 0.11.4 on playcode.io didn't fix it. But I tried upgrading our app to use 0.11.4 and issue appears to be fixed. I have no idea why it's still present on playcode.

Possibly the bug got fixed when `url` switched to `qs` instead of `querystring`. I'll close this issue, thanks!
